valhalla/midgard/openlr.h

```diff
@@ -205,23 +205,17 @@ struct LineLocation {
 
     // Offsets
     std::bitset<8> flags(attribute4);
-    poff = (index < size) && flags[6] ? lrps.front().distance * raw[index++] / 256 : 0.f;
-    noff = (index < size) && flags[5] ? std::next(lrps.rbegin())->distance * raw[index++] / 256 : 0.f;
+    poff = (index < size) && flags[6] ? raw[index++] : 0;
+    noff = (index < size) && flags[5] ? raw[index++] : 0;
   }
 
-  LineLocation(const std::vector<LocationReferencePoint>& lrps, float poff, float noff)
-      : lrps(lrps),
-        poff(lrps.at(0).distance > 0
-                 ? std::round(poff / lrps.at(0).distance * 256) / 256 * lrps.at(0).distance
-                 : 0.f),
-        noff(lrps.at(lrps.size() - 2).distance > 0
-                 ? std::round(noff / lrps.at(lrps.size() - 2).distance * 256) / 256 *
-                       lrps.at(lrps.size() - 2).distance
-                 : 0.f) {
-    if (poff > lrps.front().distance)
-      throw std::invalid_argument("Positive offset out of range");
-    if (noff > std::next(lrps.rbegin())->distance)
-      throw std::invalid_argument("Negative offset out of range");
+  LineLocation(const std::vector<LocationReferencePoint>& lrps,
+               uint8_t positive_offset_bucket,
+               uint8_t negative_offset_bucket)
+      : lrps(lrps), poff(positive_offset_bucket), noff(negative_offset_bucket) {
+    if (poff > noff) {
+      throw std::invalid_argument("Positive offset cannot be greater than the negative offset");
```

Comment:
this doesnt seem right to me. the only time poff and noff have any interplay with each other is if the `LineLocation` has only two lrps (ie is the most simple type). i think that you can only assert that:

```c++
if (lrps.size() == 2 && 255 - poff < noff)
  throw std::invalid_argument("Simple linear references cannot have combined offsets greater than the length of the reference");
```

I guess we should also throw if you only provide 1 lrp as well.

Reply:
Good spot, yeah, I didn't think of that case.  I've added both these checks.